File: Brick Sort/brick-sort.go
package bricksort

// A function to sort the algorithm using Odd Even sort
func oddEvenSort(arr []int, n int) {
	isSorted := false

	// Bubble sort on odd element
	for !isSorted {
		isSorted = false
		for i := 1; i <= n-2; i = i + 2 {
			if arr[i] > arr[i+1] {
				arr[i], arr[i+1] = arr[i+1], arr[i] // Swaps elements
				isSorted = false
			}
		}

		// Bubble sort on even element
		for i := 0; i <= n-2; i = i + 2 {
			if arr[i] > arr[i+1] {
				arr[i], arr[i+1] = arr[i+1], arr[i] // Swaps elements
				isSorted = false
			}
		}
	}

}

File: Brick Sort/BrickSort-Javascript/brickSort.js
/**
 * Brick Sort algorithm.
 * A variation of the Bubble Sort algorithm which in each iteration performs a bubble sort in two steps,
 * first on odd indexed elements, then on even indexed elements.
 * Complexity: O(N^2).
 * @param {Array} arr Input array.
 * @return {Array} Sorted array.
 */
const brickSort = (arr) => {
  // Swaps array elements at indexes i and j
  const swapElements = (arr, i, j) => {
    let temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
  };

  let sorted = false;

  while (!sorted) {
    sorted = true;
    // Loop through and sort odd indexed elements
    for (let i = 1; i < arr.length - 1; i += 2)
      if (arr[i] > arr[i + 1]) {
        swapElements(arr, i, i + 1);
        sorted = false;
      }

    // Loop through and sort even indexed elements
    for (let i = 0; i < arr.length - 1; i += 2)
      if (arr[i] > arr[i + 1]) {
        swapElements(arr, i, i + 1);
        sorted = false;
      }
  }

  return arr;
};
